returnn/tf/layers/rec.py

```diff
@@ -1043,11 +1043,14 @@ def _construct_template(self):
     from pprint import pformat
     from collections import OrderedDict
     from returnn.util.basic import StringIO
-    from returnn.tf.network import NetworkConstructionDependencyLoopException
+    from returnn.tf.network import NetworkConstructionDependencyLoopException, DataNotFound
     # The stack trace is not so interesting for these exceptions.
     skip_stack_trace_exception_types = (
       NetworkConstructionDependencyLoopException, LayerNotFound)
 
+    # These Exceptions always indicate incorrect construction, so fail directly instead of collecting them
+    fail_directly_exception_types = (DataNotFound, )
```

Comment:
Why the extra space after `,`?

Why is `LayerNotFound` not in here?

Reply:
PyCharm gives a warning if there is no whitespace.

Yes, LayerNotFound probably can be moved.

Reply:
Oh, didn't see this.